nncf/common/tensor_statistics/aggregator.py

```diff
@@ -149,3 +199,44 @@ def _process_outputs(outputs: Any) -> Dict[str, NNCFTensor]:
         :param outputs: raw model outputs
         :return: processed model outputs in Dict[str, Tensor] format
         """
+
+    @abstractmethod
+    def _get_statistics_key(self, statistics: TensorStatistic, target_point: TargetPoint) -> str:
+        """
+        Returns key of statistics.
+
+        :param statistics: Statistics value.
+        :param target_point: Statistics target point.
+        :return: Statistics key.
+        """
+
+
+class StatisticsSerializer:
+    @staticmethod
+    def load_from_file(file_name: str) -> Any:
+        """
+        Loads statistics from a gzip-compressed file.
+        :param file_name: The name of the file from which to load the statistics.
+        :return: The loaded statistics.
+        """
+        try:
+            with gzip.open(file_name, "rb") as f:
+                return pickle.load(f)
+        except FileNotFoundError:
+            raise nncf.ValidationError(f"File not found: {file_name}")
+        except (pickle.UnpicklingError, IOError):
+            raise nncf.ValidationError(f"Error loading statistics from {file_name}")
```

Comment:
I wonder what would be the behavior if statistics were saved for OV, but loaded for Torch backend? Which error would be printed?  Will it fail here or somewhere in the internals of NNCF? 
It would be great to mention in the error message that the backend for statistics is inconsistent.

Reply:
This scenario looks as an edge case for me. But I want to get opinion from @alexsu52 and @AlexKoff88. 
I think we can find the solution after the main part of PR is confirmed

Reply:
I agree with @ljaljushkin. The statistic should include backend into the key.

Reply:
I added backend info in statistics with a proper check. Please take a look https://github.com/openvinotoolkit/nncf/pull/3017/files#diff-1a0f987b2410f42c05200f78442b433da1ea428389ec3222104b47895499edbbR138
https://github.com/openvinotoolkit/nncf/pull/3017/files#diff-1a0f987b2410f42c05200f78442b433da1ea428389ec3222104b47895499edbbR227

Reply:
Added backend validation from metadata file